examples/vue/src/App.vue

```diff
@@ -0,0 +1,228 @@
+<template>
+  <div class="app">
+    <h1>Form Validation with Fonk and Vue.js 2</h1>
+    <form id="form">
+      <div>
+        <label>First Name</label>
+        <input
+          type="text"
+          placeholder="First name"
+          :value="values.firstName"
+          @input="handleInputChange('firstName')"
+          @blur="handleInputChange('firstName')"
+        />
+        <span>{{errors.firstName.message}}</span>
+      </div>
+      <div>
+        <label>Last Name</label>
+        <input
+          type="password"
+          placeholder="Last name"
+          :value="values.lastName"
+          @input="handleInputChange('lastName')"
+          @blur="handleInputChange('lastName')"
+        />
+        <span>{{errors.lastName.message}}</span>
+      </div>
+      <div>
+        <label>Age</label>
+        <input
+          type="number"
+          placeholder="Age"
+          :value="values.age"
+          @input="handleInputChange('age')"
+          @blur="handleInputChange('age')"
+        />
+        <span>{{errors.age.message}}</span>
+      </div>
+      <div class="buttons">
+        <button type="submit" @click.prevent="onValidateForm">Submit</button>
+        <button id="reset-button" type="button" @click="resetButton">Reset</button>
+      </div>
+      <pre>{{values}}</pre>
+    </form>
+  </div>
+</template>
+
+<script>
+import { createDefaultValidationResult } from '@lemoncode/fonk';
+import { formValidation } from './form-validation';
+
+const createEmptyValues = () => ({
+  firstName: '',
+  lastName: '',
+  age: '',
+});
+
+const createEmptyErrors = () => ({
+  firstName: createDefaultValidationResult(),
+  lastName: createDefaultValidationResult(),
+  age: createDefaultValidationResult(),
+});
+
+export default {
+  name: 'app',
+  data() {
+    return {
+      values: createEmptyValues(),
+      errors: createEmptyErrors(),
+    };
+  },
+  methods: {
+    handleInputChange(fieldName) {
+      const { value } = event.target;
+      this.values[fieldName] = value;
+
+      formValidation
+        .validateField(fieldName, value)
+        .then(validationResult => (this.errors[fieldName] = validationResult));
+    },
+    async onValidateForm() {
+      try {
+        const validationResult = await formValidation.validateForm(this.values);
+        this.errors = { ...this.errors, ...validationResult.fieldErrors };
```

Comment:
I think this is not necessary any more, we can replace this.error with `validationResult.fieldErrors`